### ISSUE TITLE
Question: Can this engine support having both graphical tiles and ASCII? Or is the ASCII completely baked in?

### ISSUE THREAD

**Issue by JonathanLochridge** [Lyle-Tafoya/Omega#36]
I was wondering if it would be feasible to add a graphical tileset mode to this to maybe get it some more attention?
it seems like a really cool game.

**Comment by TonberryEnjoyer** [comment]
Ascii looks perfectly fine.
I recommend using a terminal emulator with ubuntu mono or your favorite font.
Issue closed.

**Comment by wtanksleyjr** [comment]
Although the present author isn't inclined to add that to his list of tasks:

It's been done. The Mac version had a very attractive GUI version, and the
author of it was porting his engine to Windows - in fact it's a brutal fact
that the sourceforge version of the code is broken BECAUSE that port was
abandoned in the middle of the task (I don't know why it was abandoned, but
I also wasn't able to figure out what went wrong well enough to fix it).

Anyhow, right now I think one plausible path might be to try to rescue that
port, either by trying to port it to a fork of this project or by trying to
rescue the old Windows port.

https://sourceforge.net/projects/omega-roguelike/

Anyhow, if anyone wants to try to do that, I'd be very glad to help, as my
name is on that repository so I'm kind of implicated in the failure (I'm
'wtanksle'), so ping me.

-Wm

**Comment by Lyle-Tafoya** [comment]
> I was wondering if it would be feasible to add a graphical tileset mode to this to maybe get it some more attention? it seems like a really cool game.

This version of Omega uses curses for drawing, which is not made for tile graphics. However, there is a project called Necklace of the Eye (NotEye) which can be used to add tile graphics to curses roguelikes. This is what ADOM uses for adding tiles. I'm not sure how much of an undertaking this would be though.

Tiles are not something that interests me personally, so this is not something I have focused my efforts on. However, I also understand that tiles are important for a lot of people. If someone were to submit a pull request which adds support for tiles, I may consider accepting it.

**Comment by Lyle-Tafoya** [comment]
One of my long-term goals is actually to cleanup the display code to get all the curses routines wrapped by a class. If that class is carefully designed, such an abstraction could make it easier to add support for tiles without having to use Necklace of the Eye.

**Comment by wtanksleyjr** [comment]
Here's a screenshot from the Mac version I mentioned.

![macomega](https://github.com/Lyle-Tafoya/Omega/assets/1307641/f24a2bf9-a1e8-4827-b3f2-e069c29e45fd)

**Comment by Lyle-Tafoya** [comment]
That looks decent. I think it's neat how the altars all have different symbols depending on the god.

**Comment by JonathanLochridge** [comment]
I personally quite like ascii, but I am a little bit of an artist.
And my C++ is kind of meh. So I figured it might be an easier way to contribute to the project. I haven't worked too much with curses though. At least the color palette is sensible for this game. 

I also didn't know you could swap the fonts. 
I probably could make it look a little bit better to my personal tastes by doing that at least.

I am also a pretty big fan of using custom 1-bit glyphs that use the same kind of coloring system as ascii but technically aren't ascii sometimes. Particularly for stuff like walls. But it can look pretty good with other stuff too.

Having actual tile support or at least support for larger character sets is nice though.
But you can still do a decent amount graphics wise without that even.

Rather than going for some uber-fancy. Or worse a meh looking one that technically has colors.
A lot of roguelikes have tilesets that look worse than the ascii honestly. But I like it when they still have a similar feeling at least.

**Comment by MuffinManKen** [comment]
I started messing around with building Omega as a DLL that I call from the Godot game engine.  It's a horrible hack, but I did get it to render something.
<img width="592" height="797" alt="Image" src="https://github.com/user-attachments/assets/213dfa77-6950-4ae2-bdce-2f47f6144962" />

**Comment by Lyle-Tafoya** [comment]
That's neat. How is Godot rendering it?

**Comment by MuffinManKen** [comment]
> That's neat. How is Godot rendering it?

There's a curses stub that renders the tiles to a character buffer. Godot gets that buffer and loops through. Right now I'm just rendering it as the ascii but that would be easy to change.  It's pretty messy and the curses stub is a real hack. Godot sends any keyboard input to the Omega DLL for processing.

**Comment by Lyle-Tafoya** [comment]
If I understand correctly, this would prevent it from being able to distinguish between different entities represented by the same character/color combination. If so, it sounds fundamentally equivalent to using NotEye. NotEye was made for exactly this purpose and would likely be easier to use. It's what ADOM and Hydra Slayer both use for adding tiles support. https://roguetemple.com/z/noteye/

**Comment by MuffinManKen** [comment]
Yes. I was hoping deeper integration would be simpler, but Omega's code design makes it more challenging. Right now this is more of a test to see if I could.

**Comment by MuffinManKen** [comment]
Thanks for the heads up that I was heading down a blind alley. It really seems like there would be no point in continuing to tinker in that direction.

**Comment by Lyle-Tafoya** [comment]
Is it possible to access Omega's global variables from Godot? If so, it shouldn't be that hard to loop over the map and draw it with tiles

**Comment by MuffinManKen** [comment]
Not directly. Everything needs to be wrapped and exposed in a Godot-friendly way.

**Comment by MuffinManKen** [comment]
The messing around continues:

<img width="1915" height="848" alt="Image" src="https://github.com/user-attachments/assets/2c26b00c-9ab6-470d-94e0-8ee5da94e0f2" />

**Comment by Lyle-Tafoya** [comment]
This is interesting. Please continue updating us with any further developments.

**Comment by Lyle-Tafoya** [comment]
So I've been thinking about it, and although getting the game to run in Godot is cool, it also seems a bit overkill. In the near future, I'll probably just make a SDL3 tiles version of Omega. Getting quality tiles for everything will be the hard part, but the rendering with SDL should be easy.

**Comment by MuffinManKen** [comment]
Using Godot is absolutely overkill. I'm not doing it because it's the sensible choice, but because I'm having fun with it. :)
SDL3 is a great choice.

**Comment by MuffinManKen** [comment]
Also, you can generate tiles of the ascii characters so it could support ascii or graphics all with exactly the same code.
